pkg/apis/core/types.go

```diff
@@ -2693,6 +2693,12 @@ type PodSpec struct {
 	// If not specified, the pod will not have a domainname at all.
 	// +optional
 	Subdomain string
+	// If set to true, the fully qualified domain name (FQDN) of a Pod is set as hostname of its containers.
+	// In Linux containers, this means setting the FQDN in the hostname field of the kernel (the nodename field of struct utsname).
+	// In Windows containers, this means setting the COMPUTERNAME to FQDN.
+	// If a pod does not have FQDN, this has no effect.
+	// +optional
+	HostnameFQDN bool
```

Comment:
There's a general convention of optionals always being pointers.  It is not 100% followed.  In this case it is kind of tiresome because it adds very little value to the system (since false is correct as the default).  @liggitt or @lavalamp - I feel like I am swimming upstream on this - are others enforcing this convention?

Reply:
Not making it a pointer means that users who construct their objects with the go typed objects cannot express the difference between "I want this off" and "I don't care". Doesn't matter for pod, but it could matter for the places where there are pod templates. SSA uses this bit to determine whether or not to give other users changing the field conflicts.

I don't think it's too likely that a completely separate actor would want to own this field vs other fields in pod templates (e.g., doesn't make sense in the way HPA -> replicas or VPA -> resource request). So it might not be super important.

Also users who directly construct the object in yaml/json do not have this problem.

Reply:
@thockin updated to use *bool

Reply:
Ah sorry for lack of clarity! My overall point was that it's probably OK to leave it a non pointer in this case. But of course I'm fine with it being a pointer.